packages/web3-errors/src/errors/transaction_errors.ts

```diff
@@ -84,25 +85,54 @@ export class RevertInstructionError extends BaseWeb3Error {
 	}
 }
 
-export class TransactionRevertError extends BaseWeb3Error {
+export class TransactionRevertError<ReceiptType = TransactionReceipt> extends BaseWeb3Error {
 	public code = ERR_TX_REVERT_TRANSACTION;
 
 	public constructor(
 		public reason: string,
 		public signature?: string,
-		public receipt?: TransactionReceipt,
+		public receipt?: ReceiptType,
+		public data?: string,
 	) {
-		super(
-			`Transaction has been reverted by the EVM:\n ${JSON.stringify(receipt, undefined, 2)}`,
-		);
+		super(reason);
```

Comment:
Removed `Transaction has been reverted by the EVM:\n ` because the error is already called `TransactionRevertError` and the prefix doesn't add anymore benefit to the user. As mentioned above with `ResponseError`, removing the prefix allows the actual revert reason to be more accessible to the end user to be used within their error handling code

Also removed `JSON.stringify(receipt, undefined, 2)` because receipts are not always available when this error is thrown, makes more sense just to leave it as a property of error object